run-python/python.go

```diff
@@ -0,0 +1,266 @@
+package runner
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/stoic-cli/stoic-cli-core"
+)
+
+const (
+	getPipCacheKey = "run-python/get-pip.py"
+	getPipURL      = "https://bootstrap.pypa.io/get-pip.py"
+
+	pipCacheBase     = ".pip-cache"
+	readyBase        = ".ready"
+	requirementsBase = ".requirements"
+
+	pipRequirements = "" +
+		"pip==10.0.1 --hash=sha256:717cdffb2833be8409433a93746744b59505f42146e8d37de6c62b430e25d6d7\n" +
+		"setuptools==39.1.0 --hash=sha256:0cb8b8625bfdcc2d43ea4b9cdba0b39b2b7befc04f3088897031082aa16ce186\n" +
+		"virtualenv==15.2.0 --hash=sha256:e8e05d4714a1c51a2f5921e62f547fcb0f713ebbe959e0a7f585cc8bef71d11f\n" +
+		"wheel==0.31.0 --hash=sha256:9cdc8ab2cc9c3c2e2727a4b67c22881dbb0e1c503d592992594c5e131c867107\n" +
+		""
+)
+
+func getPipScript(cache stoic.Cache) (string, error) {
+	var err error
+
+	script, err := ioutil.TempFile("", "get-pip-*.py")
+	if err != nil {
+		return "", fmt.Errorf(
+			"unable to set up temp file for get-pip.py: %v", err)
+	}
+	defer func() {
```

Comment:
This defer should come before the previous if. Also, why the if test inside the defer? Don't we also want to remove the temp file? We could also consider explicitly ignoring the errors returned by the os.Remove and script.Close() with `_ = ...`

Reply:
This can't come before the previous `if`, because at that point the temp file creation failed, so there's nothing to `Remove()` or `Close()`.

What the `if` inside the function is meant to do is delete the temporary file if we're returning an error. Otherwise, the file stays and is deleted by the caller, who will use the file to setup `pip` and friends.